### DISCUSSION TITLE
How to grant a UDF to common role or user.

### DISCUSSION THREAD

**Discussion by Young6688** [apache/doris#41269] [A - General / Q&A] [answered]
How to grant a UDF to common role or user, this shows an error when I called a UDF logging as common user, I also grant the UDF to the role by using "grant SELECT_PRIV on internal.db.UDF to role 'readonly';"
![image](https://github.com/user-attachments/assets/9c595784-c671-4bf4-8138-69d753f0807f)

**Comment by ixzc** [comment]
you should grant these privilege, which depends on your UDF function
<img width="987" alt="image" src="https://github.com/user-attachments/assets/6b070b46-867b-462f-a9df-3f719bbdcf5d">

**Reply by Young6688** [reply]
Thank you for your reply. 
I have granted "normal" workload group to the role by using "grant USAGE_PRIV on WORKLOAD GROUP 'normal' to role 'readonly'", it doesn't work.

**Reply by Young6688** [reply]
I think the same problem hanppend on my server, is it a bug?
https://ask.selectdb.com/questions/D1hJ1/zi-ding-yi-function-dang-doris-zhong-qi-yi-hou-fa-xian-bu-fen-yong-hu-mei-you-fang-wen-zi-ding-yi-function-de-quan-xian

**Comment by Young6688** [comment, marked as answer]
I solve this problem by creating global funtion and recreate it every time i restart server.
